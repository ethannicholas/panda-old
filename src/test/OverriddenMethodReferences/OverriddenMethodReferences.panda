class Super {
    method hello() {
        Console.writeLine("\{class.name} calling Super.hello()!")
    }
}

class Sub : Super {
    @override
    method hello() {
        Console.writeLine("\{class.name} calling Sub.hello()!")
    }
}

def sub := new Sub()
sub.hello()

def hello := Super::hello
hello(sub)
hello(new Super())
