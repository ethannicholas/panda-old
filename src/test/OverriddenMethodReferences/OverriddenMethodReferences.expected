Sub calling Sub.hello()!
Sub calling Sub.hello()!
Super calling Super.hello()!
